src/java.base/share/classes/java/lang/IO.java

```diff
@@ -0,0 +1,196 @@
+     * readln() instead of the readln(prompt) overload. It's also important to
+     * flush after print() in case the user is relying on print() to emit output
+     * as sort of a progress indicator.
+     *
+     * We rely on System.out to have autoflush enabled, which flushes after every
+     * println() call, so we needn't flush again. We flush unconditionally after
+     * calls to print(). Since System.out is doing a lot of flushing anyway, there
+     * isn't much point trying to make this conditional, for example, only if
+     * stdout is connected to a terminal.
+     */
+
+    private IO() {
+        throw new Error("no instances");
+    }
+
+    /**
+     * Writes a string representation of the specified object and then writes
+     * a line separator to the standard output.
+     *
+     * <p> The effect is as if {@link java.io.PrintStream#println(Object) println(obj)}
+     * had been called on {@code System.out}.
+     *
+     * @param obj the object to print, may be {@code null}
+     */
+    public static void println(Object obj) {
+        System.out.println(obj);
+    }
+
+    /**
+     * Writes a line separator to the standard output.
+     *
+     * <p> The effect is as if {@link java.io.PrintStream#println() println()}
+     * had been called on {@code System.out}.
+     */
+    public static void println() {
+        System.out.println();
+    }
+
+    /**
+     * Writes a string representation of the specified object to the
+     * standard output.
+     *
+     * <p> The effect is as if {@link java.io.PrintStream#print(Object) print(obj)}
+     * had been called on {@code System.out}.
+     *
+     * @param obj the object to print, may be {@code null}
+     */
+    public static void print(Object obj) {
+        var out = System.out;
+        out.print(obj);
+        out.flush();
+    }
+
+    /**
+     * Reads a single line of text from the standard input.
+     * <p>
+     * One line is read from the decoded input as if by
+     * {@link java.io.BufferedReader#readLine() BufferedReader.readLine()}
+     * and then the result is returned.
+     * <p>
+     * If necessary, this method first sets up charset decoding, as described in
+     * above in the class specification.
+     *
+     * @return a string containing the line read from the standard input, not
+     * including any line separator characters. Returns {@code null} if an
+     * end of stream has been reached without having read any characters.
+     *
+     * @throws IOError if an I/O error occurs
+     */
+    public static String readln() {
+        try {
+            return reader().readLine();
+        } catch (IOException ioe) {
+            throw new IOError(ioe);
+        }
+    }
+
+    /**
+     * Writes a prompt and then reads a line of input.
+     * <p>
+     * Writes a prompt as if by calling {@code print}, and then reads a single
+     * line of text as if by calling {@link readln readln()}.
```

Comment:
Is this intended to render like `readln` or `readln()`? I always recommend prefixing member references with `#` like `#readln()`.

Reply:
I added `#` prefixes. I had intended `readln()` to indicate that this referred to the no-arg overload of `readln()` but I think that's too subtle so I removed the parens. But one has to look at the link to determine that it refers to the no-arg readln and not the one-arg readln (which is this method). I also changed the `print` to a link as well.